src/views/List.jsx

```diff
@@ -1,15 +1,69 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	// state for input
+	const [input, setInput] = useState('');
+	const [searchData, setSearchData] = useState(data);
+	const [isValid, setIsValid] = useState(false);
+	const [searchLength, setSearchLength] = useState(1);
+	// Handle event
+	const handleInputChange = (e) => {
+		setInput(e.target.value);
+	};
+
+	const handleInputClear = () => {
+		setInput('');
+	};
+
+	useEffect(() => {
+		const searchRegex = new RegExp(input, 'i');
+		const filteredData = data.filter((item) => searchRegex.test(item.name));
+		setSearchData(filteredData);
+	}, [data, input]);
+
+	useEffect(() => {
+		if (searchData.length > 0) {
+			setIsValid(true);
+			setSearchLength(50);
+		} else {
+			setIsValid(false);
+			setSearchLength(0);
+		}
+	}, [searchData, data]);
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
-			<ul>
-				{data &&
-					data.map((item, index) => <ListItem key={index} name={item.name} />)}
-			</ul>
+
+			<label htmlFor="input">
+				{' '}
+				Search for your item:
+				<input
+					type="text"
+					id="input"
+					value={input}
+					onChange={handleInputChange}
+					maxLength={searchLength}
+				/>
+			</label>
+			{input.length > 0 ? (
+				<button onClick={handleInputClear} value={input}>
+					x
+				</button>
+			) : null}
+			{isValid ? (
+				<ul>
+					{searchData &&
+						searchData.map((item, index) => (
+							<ListItem key={index} name={item.name} />
```

Comment:
Your PR didn't change this code, but consider changing the key to `item.id` instead and because each item already has a unique identifier.
https://react.dev/learn/rendering-lists